### PR TITLE
feat: Enable checktyle warning and Java Format

### PR DESCRIPTION
**Pull request by dricazenck**
* Enable project default format in intelliJ
* Enable checkstyle warnings for src and test folders
* Update readme to include the necessary steps to setup project in Intellij

Related to #10

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,67 +1,103 @@
...
 
-* Create Jar
-
-```shell
-./gradlew clean bootJar
-```
+## Setup
 
-* Start Spring Boot Application:
+### JAVA 21 with SDKMAN
```
Comment (goelsonali) [inline README.md:16]: Good to include gradle version used for the project

File: README.md [hunk 2]
```diff
@@ -1,67 +1,103 @@
...
 
-## Set the default Java version for your system:
+Set the default Java version for your system:
 
 * To set the newly installed Java version as the default:
+
 ```shell
 sdk default java 21.0.2-open
 ```
+
```
Comment (goelsonali) [inline README.md:57]: Either remove this statement how to verify your java version or add the java - version command 
Which i can see has been removed below.

File: README.md (hunk 1 above)
Comment (womencodingcommunity) [reply README.md:16]: But that is not something you have to install, it is included into the wrapper itself. I don't think we should include here.

File: README.md (hunk 2 above)
Comment (womencodingcommunity) [reply README.md:57]:
```suggestion

* Check the java 

```shell
java -version
```
```

**Review by womencodingcommunity — APPROVED** [review]

**Pull request merged**